### ISSUE TITLE
Training not converge

### ISSUE THREAD

**Issue by xiaoliangbai** [vinvino02/GLPDepth#7]
Firstly, thanks for the nice paper and open source project!

     I am trying to reproduce the training process. I am using NYUDepthV2 dataset from Kaggle, it should be identical, just filename is different. Nothing else changed, only dataloader was refactored as 'kaggle' and it loads csv files for training and test. 

  As you can see from the logs.txt (copied below) in the 25 epochs,  the silog is reducing (as expected), but RMSE jumps up dramatically around epoch 15. Maybe it was caused by the poly LR schedule around half_epoch. Anyway, it seems with the default settings, I could not reproduce the best-trained model using NYU depth v2 dataset.  Is there anything I should try/modify?

Thanks a lot!
Shawn

Copy of the logs.txt:
==========================================================

gpu_or_cpu:gpu,	
data_path:./datasets/,	
dataset:kaggle,	
exp_name:local_kaggle,	
batch_size:12,	
workers:8,	
max_depth:10.0,	
max_depth_eval:10.0,	
min_depth_eval:0.001,	
do_kb_crop:1,	
kitti_crop:None,	
epochs:25,	
lr:0.0001,	
crop_h:448,	
crop_w:576,	
log_dir:./logs,	
val_freq:1,	
save_freq:10,	
save_model:False,	
save_result:False,	

Epoch: 001 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9725     2.6005     2.8542     3.6123     1.5674     2.5565 
====================================================================================================

Epoch: 002 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9749     2.6100     2.8581     3.7024     1.6068     2.6198 
====================================================================================================

Epoch: 003 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9734     2.6032     2.8548     3.6429     1.5810     2.5776 
====================================================================================================

Epoch: 004 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9740     2.6057     2.8561     3.6663     1.5912     2.5941 
====================================================================================================

Epoch: 005 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9737     2.6048     2.8557     3.6555     1.5865     2.5866 
====================================================================================================

Epoch: 006 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9734     2.6035     2.8551     3.6427     1.5810     2.5775 
====================================================================================================

Epoch: 007 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9734     2.6029     2.8547     3.6415     1.5805     2.5766 
====================================================================================================

Epoch: 008 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9733     2.6028     2.8547     3.6377     1.5788     2.5739 
====================================================================================================

Epoch: 009 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9734     2.6035     2.8551     3.6432     1.5812     2.5778 
====================================================================================================

Epoch: 010 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9737     2.6048     2.8557     3.6541     1.5860     2.5854 
====================================================================================================

Epoch: 011 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9734     2.6030     2.8547     3.6412     1.5804     2.5763 
====================================================================================================

Epoch: 012 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9733     2.6027     2.8545     3.6391     1.5795     2.5748 
====================================================================================================

Epoch: 013 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9735     2.6036     2.8551     3.6449     1.5820     2.5789 
====================================================================================================

Epoch: 014 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9736     2.6043     2.8555     3.6496     1.5840     2.5822 
====================================================================================================

Epoch: 015 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0071     0.0228     0.0606     3.5239    27.9788     7.3483     1.4549     0.6088     1.0641 
====================================================================================================

Epoch: 016 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0071     0.0228     0.0606     3.5245    27.9877     7.3496     1.4550     0.6088     1.0642 
====================================================================================================

Epoch: 017 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0071     0.0228     0.0605     3.5245    27.9878     7.3496     1.4550     0.6088     1.0642 
====================================================================================================

Epoch: 018 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0071     0.0228     0.0605     3.5245    27.9878     7.3496     1.4550     0.6088     1.0642 
====================================================================================================

Epoch: 019 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0071     0.0228     0.0605     3.5245    27.9878     7.3496     1.4550     0.6088     1.0642 
====================================================================================================

Epoch: 020 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0071     0.0228     0.0605     3.5245    27.9878     7.3496     1.4550     0.6088     1.0642 
====================================================================================================

Epoch: 021 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0071     0.0228     0.0605     3.5245    27.9878     7.3496     1.4550     0.6088     1.0642 
====================================================================================================

Epoch: 022 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0071     0.0228     0.0605     3.5245    27.9878     7.3496     1.4550     0.6088     1.0642 
====================================================================================================

Epoch: 023 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0071     0.0228     0.0605     3.5245    27.9878     7.3496     1.4550     0.6088     1.0642 
====================================================================================================

Epoch: 024 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0071     0.0228     0.0605     3.5245    27.9878     7.3496     1.4550     0.6088     1.0642 
====================================================================================================

Epoch: 025 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0071     0.0228     0.0605     3.5245    27.9878     7.3496     1.4550     0.6088     1.0642 
====================================================================================================

**Comment by vinvino02** [comment]
Hello. I have downloaded docker image and github code and trained again with the official dataset to reproduce the results , and I confirmed that the training loss converges. 
Could you tell me where I can get the dataset you used if I can approach it? or could you try to train the model with the official dataset?

**Comment by xiaoliangbai** [comment]
Thanks for the response.
I downloaded the NYU depth v2 dataset from Kaggle at:
https://www.kaggle.com/soumikrakshit/nyu-depth-v2

I will try the original dataset once I have downloaded it.

Thanks!

**Comment by xiaoliangbai** [comment]
Hello vinvino02;

 Which toolkit was used to convert *dump files to RGB and depth? I want to make sure the converted dataset is matching what you are using.
 
 Thanks!

**Comment by xiaoliangbai** [comment]
Confirmed that with docker and bts NYU depth V2 dataset, the training converges very fast; This issue can be closed.
Thanks!

**Comment by hustXYY** [comment]
Hi! May I ask how you solved this problem? Is the changing to BTS NYUv2 dataset that works? 
And what is your CUDA and pytorch environment version?
Thanks!

**Comment by xiaoliangbai** [comment]
Please try to use docker image and bts NYU v2 dataset. I can reproduce the result with this setting. 
Thanks

**Comment by JHCheng111** [comment]
> Please try to use docker image and bts NYU v2 dataset. I can reproduce the result with this setting. Thanks

Thanks for your advance! 
I'm sorry, I have to ask you one more question: I don't know what is the docker image and the data set I used was bts NYUv2 data set in fact. But I can't reproduce the result, so the docker image is necessary, right?  Why the metrics result became abnormal from the 11th epoch(i.e. d1、d2、d3=0, the logs I've pasted it in the comments above)? Is it a data set problem?
Thank you very much for your help!
